### DISCUSSION TITLE
[Feature] Support HIF Extension for heif format images

### DISCUSSION THREAD

**Discussion by EYHN** [immich-app/immich#6131] [Feature Request]
### The feature

Heif format compressed images created by Sony and fujifilm cameras are saved by the camera with a `.hif` extension name. It would be helpful if immich could support recognizing `hif` extension as heif file.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by haossu** [comment]
https://github.com/immich-app/immich/pull/5197 like this pr？

**Comment by mmomjian** [comment]
Supported

**Reply by JU3RG** [reply]
Running 3.2.1. I'd like to check what "supported" covers here, because I'm seeing `.hif` and `.heic` behave differently downstream of ingestion.

My camera (Fujifilm X100VI) writes HEIF to the card as `.HIF`. Those files upload fine, generate thumbnails and previews, and EXIF is extracted correctly — so ingestion clearly works.

However, with "Load original image" enabled in the iOS app, a `.HIF` asset will not serve the original (pinch-zoom stays at preview resolution). If I rename the same file to `.heic` and re-upload, the original is served normally, and it behaves identically to HEICs from my iPhone.

The rename is a plain `mv` — no transcode, byte-identical contents. Extension is the only variable.

Is `.hif` support intended to cover only upload/processing, or should it be treated identically to `.heic` throughout? If the latter, I'm happy to open a separate issue with full details — just wanted to check here first since this discussion is closed as supported.
